### PR TITLE
Add Normalization functionality to Linux

### PR DESCRIPTION
**Pull request by tarekgh**
This functionality will be used in different places (e.g. string class)

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @jkotas @danmosemsft

File: src/mscorlib/src/System/Text/Normalization.Unix.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        /// unlike Windows (which uses the ERROR_NO_UNICODE_TRANSLATION error value to signal an error).
+        ///
+        /// We walk the string ourselves looking for these bad sequences so we can continue to throw
+        /// ArgumentException in these cases.
+        /// </summary>
+        private static bool HasInvalidUnicodeSequence(string s)
+        {
+            for (int i = 0; i < s.Length; i++)
+            {
+                char c = s[i];
```
Comment (jkotas) [inline src/mscorlib/src/System/Text/Normalization.Unix.cs]: Having early out `if (c < '\ud800') continue` would make this look number of times faster.

Comment (tarekgh) [inline src/mscorlib/src/System/Text/Normalization.Unix.cs]: good idea. I'll do that.

---
In reply to: [81041918](https://github.com/dotnet/coreclr/pull/7404#discussion_r81041918) [](ancestors = 81041918)

**Comment by tarekgh** [comment]
> The duplicate implementation in corefx can be removed once the dust settles, correct? We should make sure to have issue to track the cleanup.

I have created the issue:
https://github.com/dotnet/corefx/issues/12168

Thanks @jkotas for your review.

**Comment by tarekgh** [comment]
https://github.com/dotnet/corefx/issues/11617

**Pull request merged**